code/modules/events/solarflare.dm

```diff
@@ -0,0 +1,29 @@
+/datum/event/solar_flare
+	startWhen = 10
+	endWhen = 500 // = ~500x2 seconds or ~16m40s
+
+/datum/event/solar_flare/announce()
+	GLOB.event_announcement.Announce("A solar flare has been detected on collision course with the station.", "Flare Alert", new_sound = 'sound/AI/attention.ogg')
```

Comment:
Maybe you could include a warning? Like how radstorms tell you to seek cover, maybe it should say "suspend all EVA activity until further notice" or something so people unfamiliar with the event don't get blindsided.

Reply:
I've debated doing that, but:
1) It would completely negate the point of the event, because people would de-hotwire solars before it kicked in.
2) The space damage is not very big - you have enough time to get inside before it cripples you.

Reply:
I ended up giving it a 30 second telegraphing/advanced warning. I'm really skeptical that this is a good idea, but I'll try it. In the event that people leave solars hotwired, and just de-hotwire them when they get the alert... I'm going to remove the warning completely.